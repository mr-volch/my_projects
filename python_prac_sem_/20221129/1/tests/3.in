class C(metaclass=dump):
    def __init__(self, val, yt=3):
        self.val = val
        self.yt = yt


    def kiz(self, other, another, this=2, those=None):
        return self.val + other + (another or self.val) + self.yt + this + (those or this)

c = C(10,17)
print(c.kiz(9,10,this=890))
print(c.kiz(9,10))

d = C(89)
print(d.kiz(1,2,3,4))
print(d.kiz(other = 1, another = 2, this = 3, those = False))

