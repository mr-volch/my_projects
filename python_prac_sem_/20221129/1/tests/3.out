__init__: (10, 17), {}
kiz: (9, 10), {'this': 890}
1826
kiz: (9, 10), {}
50
__init__: (89,), {}
kiz: (1, 2, 3, 4), {}
102
kiz: (), {'other': 1, 'another': 2, 'this': 3, 'those': False}
101
